libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Component that translates Solidity code into Yul at statement level and below.
+ */
+
+#include <libsolidity/codegen/ir/IRGeneratorForStatements.h>
+
+#include <libsolidity/codegen/ir/IRGenerationContext.h>
+#include <libsolidity/codegen/YulUtilFunctions.h>
+
+#include <libdevcore/StringUtils.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::solidity;
+
+bool IRGeneratorForStatements::visit(VariableDeclarationStatement const& _varDeclStatement)
+{
+	for (auto const& decl: _varDeclStatement.declarations())
+		if (decl)
+			m_context.addLocalVariable(*decl);
+
+	solUnimplementedAssert(_varDeclStatement.declarations().size() == 1, "");
```

Comment:
This could be moved into the ``if`` case below, right? The ``else`` will be just fine for multiple declarations already.